### PR TITLE
Add workflow for github CI

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: .github/workflows/packcheck.yml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  pull_request:
+
+
+#-----------------------------------------------------------------------------
+# Build matrix
+#-----------------------------------------------------------------------------
+
+jobs:
+  build:
+    name: >-
```
Comment (pranaysashank) [inline .github/workflows/packcheck.yml:25]: what this symbol `>-` do

File: .github/workflows/packcheck.yml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+          #   # Please change the repo token to your repo token
+          #   pack_options: >-
+          #     CABAL_PROJECT=cabal.project.coveralls
+          #     COVERALLS_OPTIONS="--repo-token=YvxdfMiOmfq0limz6nLd4DxiC1RBQJ4dA --coverage-mode=StrictlyFullLines --exclude-dir=test test"
+          #   runner: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-haskell@v1
```
Comment (pranaysashank) [inline .github/workflows/packcheck.yml:157]: we can replace it with haskell/actions/setup@v1

File: .github/workflows/packcheck.yml [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+          #   runner: macos-latest
+
+          - name: hlint
+            command: cabal-v2
+            runner: ubuntu-latest
+            pack_options: >-
+              HLINT_OPTIONS="lint"
+              HLINT_TARGETS="src"
+
+          # IMPORTANT: Please change the --repo-token to your repo token
```
Comment (pranaysashank) [inline .github/workflows/packcheck.yml:139]: This doesn't work, there is no proper support for github actions in hpc coveralls

File: .github/workflows/packcheck.yml [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+      # ------------------------------------------------------------------------
+      PACKCHECK_COMMAND: ${{ matrix.command }} ${{ matrix.pack_options }}
+
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # The reason we have name here is to force an additional config instead
+        # of adding to an existing config while adding additional configs.
+        # Look at "jobs.<job_id>.strategy.matrix" in github actions to learn
```
Comment (pranaysashank) [inline .github/workflows/packcheck.yml:95]: Better to also link to the webpage

File: .github/workflows/packcheck.yml (hunk 1 above)
Comment (adithyaov) [reply .github/workflows/packcheck.yml:25]: Similar to `unwords . unlines`

File: .github/workflows/packcheck.yml (hunk 3 above)
Comment (adithyaov) [reply .github/workflows/packcheck.yml:139]: I did run all the CI's before commenting them. All of them seem to succeed. When you say it won't work does it mean it gives a false positive?

Comment (adithyaov) [reply .github/workflows/packcheck.yml:139]: Ping @pranaysashank

File: .github/workflows/packcheck.yml [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+      # version.
+      #STACKVER: "1.6.5"
+      #STACK_UPGRADE: "y"
+      #RESOLVER: "lts-12"
+
+      # ------------------------------------------------------------------------
+      # cabal options
+      # ------------------------------------------------------------------------
+      CABAL_CHECK_RELAX: y
+      CABAL_NO_SANDBOX: y
```
Comment (harendra-kumar) [inline .github/workflows/packcheck.yml:62]: This can be removed now.

File: .github/workflows/packcheck.yml [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        # of adding to an existing config while adding additional configs.
+        # Look at <https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix>
+        # for more info about adding matrix elements.
+        # Adding any element to the list will increase the number of matrix
+        # elemets proportional to the cross product.
+        name: [ci]
+        # Uncomment GHC versions you want the CI to run on.
+        ghc_version:
+          [ 8.10.1
+          # , 8.8.3
```
Comment (harendra-kumar) [inline .github/workflows/packcheck.yml:103]: Use latest versions i.e. 8.10.3 and 8.8.4.

**Review by harendra-kumar — APPROVED** [review]
I am not familiar with github workflow. Looks good to me as long as it works.

File: .github/workflows/packcheck.yml (hunk 5 above)
Comment (adithyaov) [reply .github/workflows/packcheck.yml:62]: Right, these should be removed in other ci files as well.

File: .github/workflows/packcheck.yml (hunk 3 above)
Comment (adithyaov) [reply .github/workflows/packcheck.yml]: I'm removing hpc-coveralls for now in github CI.

**Pull request merged**